Content.Server/Drone/DroneSystem.cs

```diff
@@ -0,0 +1,145 @@
+        {
+            if (args.Component.IsDead())
+            {
+                foreach (var item in drone.ToolUids)
+                {
+                        EntityManager.DeleteEntity(item);
+                }
+                _explosions.SpawnExplosion(uid);
+                EntityManager.DeleteEntity(uid);
+            }
+        }
+
+        private void OnDisarmAttempt(EntityUid uid, DroneComponent drone, DisarmAttemptEvent args)
+        {
+            EntityManager.TryGetComponent<HandsComponent>(args.TargetUid, out var hands);
+            var item = hands.GetActiveHandItem;
+            if (EntityManager.TryGetComponent<DroneToolComponent>(item?.Owner, out var itemInHand))
+            {
+            args.Cancel();
+            }
+        }
+
+        private void OnMindAdded(EntityUid uid, DroneComponent drone, MindAddedMessage args)
+        {
+            EntityManager.TryGetComponent<TagComponent>(uid, out var tagComp);
+            UpdateDroneAppearance(uid, DroneStatus.On);
+            tagComp.AddTag("DoorBumpOpener");
+            _popupSystem.PopupEntity(Loc.GetString("drone-activated"), uid, Filter.Pvs(uid));
+
+            if (drone.alreadyAwoken == false)
+            {
+                var spawnCoord = Transform(uid).Coordinates;
+
+                if (drone.Tools.Count == 0) return;
+
+                if (EntityManager.TryGetComponent<HandsComponent>(uid, out var hands))
+                {
+                   foreach (var entry in drone.Tools)
+                    {
+                        var item = EntityManager.SpawnEntity(entry.PrototypeId, spawnCoord);
+                        EntityManager.AddComponent<DroneToolComponent>(item);
+                        hands.PutInHand(item);
+                        drone.ToolUids.Add(item);
+                    }
+                }
+
+                drone.alreadyAwoken = true;
+            }
+        }
+
+        private void OnMindRemoved(EntityUid uid, DroneComponent drone, MindRemovedMessage args)
+        {
+            EntityManager.TryGetComponent<TagComponent>(uid, out var tagComp);
+            UpdateDroneAppearance(uid, DroneStatus.Off);
+            tagComp.RemoveTag("DoorBumpOpener");
+            EntityManager.EnsureComponent<GhostTakeoverAvailableComponent>(uid);
+        }
+
+        private void OnDropAttempt(EntityUid uid, DroneComponent drone, DropAttemptEvent args)
+        {
+            EntityManager.TryGetComponent<HandsComponent>(uid, out var hands);
+            var item = hands.GetActiveHandItem;
+            if (EntityManager.TryGetComponent<DroneToolComponent>(item?.Owner, out var itemInHand))
+            {
+            args.Cancel();
+            }
+        }
+
+        private void OnUnequipAttempt(EntityUid uid, DroneComponent drone, IsUnequippingAttemptEvent args)
+        {
+            if (EntityManager.TryGetComponent<DroneToolComponent>(args.UnEquipTarget, out var droneTool))
```

Comment:
```suggestion
            if (TryComp<DroneToolComponent>(args.UnEquipTarget, out var droneTool))
```

you can do this everywhere else this is used if its possible